src/base_alloc/base_alloc_global.c

```diff
@@ -82,41 +83,121 @@ static int size_to_idx(size_t size) {
     return index;
 }
 
-void *umf_ba_global_alloc(size_t size) {
+// stores metadata just before 'ptr' and returns beginning of usable
+// space to the user. metadata consists of 'size' that is the allocation
+// size and 'offset' that specifies how far is the returned ptr from
+// the origin ptr (used for aligned alloc)
+static void *transform_ptr(void *ptr, size_t size, size_t alignment) {
```

Comment:
I think the name "transform_ptr" is too generic - maybe better would be an "add_metadata"? or a pair of "pack/unpack_ptr"?

Reply:
Done.